client/i18n/language-detector.test.ts

```diff
@@ -0,0 +1,22 @@
+import { TranslationLanguage } from '../../common/i18n'
+import { getBestLanguage } from './language-detector'
+
+describe('client/i18n/language-detector', () => {
+  test('returns the english language if none of the given languages are supported', () => {
+    const lng = getBestLanguage(['INVALID_LANGUAGE'])
+
+    expect(lng).toBe(TranslationLanguage.English)
+  })
+
+  test('returns the earliest matching language without dialect', () => {
+    const lng = getBestLanguage(['en-US', 'kr'])
```

Comment:
One day I am going to get you to stop writing `'kr'` for Korean :P It's `'ko'` (and you probably want `'ko-KR'` for this to be a realistic test)